### PR TITLE
Permit rows with repeated fields and record fields to be inserted usi…

### LINKED ISSUE #325: Support record fields in BigQuery

### PR DESCRIPTION
**Pull request by jskeet**
…ng the InsertRow type

Additionally, some tests to check that record fields and repeated fields work as expected...
namely that you can't really retrieve them in a query, but you can work with "bits" of record
fields, and repeated fields are flattened.

Fixes #325 and #324 as far as we can.

### CODE REVIEW

File: src/Google.Bigquery.V2/InsertRow.cs [hunk 1]
```diff
@@ -170,16 +189,37 @@ private static object ConvertRowValue(object value)
...
+            {
+                return;
+            }
+            Type type = value.GetType();
+            if (ValidSingleTypes.Contains(type))
+            {
+                return;
+            }
+            TypeInfo typeInfo = type.GetTypeInfo();
+            if (ValidRepeatedTypes.Any(ti => ti.IsAssignableFrom(typeInfo)))
```
Comment (chrisdunelm) [inline src/Google.Bigquery.V2/InsertRow.cs]: What happens with an enumerable of a nullable valid type?

Comment (jskeet) [inline src/Google.Bigquery.V2/InsertRow.cs]: Good question. I don't believe Bigquery will actually allow a value of a repeated field to be null, but I could be wrong. Will give it a try. (We won't spot that in the case of a `string[]` of course, but the server will reject it.) If it *is* permitted then yes, we should support that...

File: test/Google.Bigquery.V2.IntegrationTests/personsData.json [hunk 2]
```diff
@@ -0,0 +1,3 @@
+{ "fullName": "John Doe", "age": 22, "gender": "Male", "phoneNumber": { "areaCode": "206", "number": "1234567"}, "children": [{ "name": "Jane", "gender": "Female", "age": "6"}, {"name": "John", "gender": "Male", "age": "15"}], "citiesLived": [{ "place": "Seattle", "yearsLived": ["1995"]}, {"place": "Stockholm", "yearsLived": ["2005"]}]}
```
Comment (chrisdunelm) [inline test/Google.Bigquery.V2.IntegrationTests/personsData.json]: Would it be better to put this in a "resources" (or something) directory?

**Comment by chrisdunelm** [comment]
Generally LGTM

File: src/Google.Bigquery.V2/InsertRow.cs (hunk 1 above)
Comment (jskeet) [inline src/Google.Bigquery.V2/InsertRow.cs]: Okay, have now tested - null values are ignored. I think it's reasonable not to support `List<int?>` etc at that point. I've updated the test to include a null tag and show that it's not included in the results, and I've updated the docs.

File: test/Google.Bigquery.V2.IntegrationTests/personsData.json (hunk 2 above)
Comment (jskeet) [inline test/Google.Bigquery.V2.IntegrationTests/personsData.json]: I think it's okay where it is - while we *could* create a resources directory, I don't think I've seen other projects doing that... typically the resources are alongside source code.

**Comment by chrisdunelm** [comment]
LGTM

**Pull request merged**